course/templates/course/course_base.html

```diff
@@ -78,12 +78,12 @@ <h4>{% trans "Calendar" %}</h4>
 		{% if instance.is_past and instance.has_enrollment_closed %}
 			<div class="alert sticky-alert alert-default">
 				{% translate "COURSE_HAS_ALREADY_ENDED" %}
-				{% if later_instance %}
+				{% if instance.later_instance and instance.later_instance|is_visible_to:user %}
```

Comment:
`instance.later_instance` runs a database query each time you use the property. I counted six calls in this if branch. Consider using either `@cached_property` in the Python class or the `with` tag in the HTML template for storing the `later_instance` value.

https://docs.djangoproject.com/en/3.2/ref/utils/#django.utils.functional.cached_property
https://docs.djangoproject.com/en/3.2/ref/templates/builtins/#with

Reply:
Fixed!